### PR TITLE
Swagger configuration

### PR DESCRIPTION
**Pull request by radoslavvlaskovski**
The swagger config file used to generate the swagger documentation. 
The api functions are annotated to be able to add descriptions and notes to the document. 
By default the swagger config file is excluded.

### CODE REVIEW

**Comment by lorenzotomasini** [comment]
How can i enable the generation of swagger documentation?

**Comment by radoslavvlaskovski** [comment]
Now the api is parsed in swagger only if on build a parameter called swagger is passed:
e.g. ./gradlew build -Pswagger

File: api/src/main/java/org/openbaton/nfvo/api/admin/RestKeys.java [hunk 1]
```diff
@@ -32,13 +33,7 @@
...
 import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestHeader;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.ResponseStatus;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
Comment (lorenzotomasini) [inline api/src/main/java/org/openbaton/nfvo/api/admin/RestKeys.java]: Same thing for static imports

File: api/src/main/java/org/openbaton/nfvo/api/admin/RestKeys.java [hunk 2]
```diff
@@ -54,6 +49,11 @@
    *
    * @param key object containing the key which needs to be uploaded
    */
+  @ApiOperation(
+    value = " Importing a Key",
+    notes =
+        "Pass the Key as JSON content in the Request Body. The Key should contain the name and public key."
```
Comment (lorenzotomasini) [inline api/src/main/java/org/openbaton/nfvo/api/admin/RestKeys.java]: Is there a way to get the string values from a config file?

File: api/src/main/java/org/openbaton/nfvo/api/admin/RestVimInstances.java [hunk 3]
```diff
@@ -100,6 +98,7 @@ public void delete(
    *
    * @return List<Datacenter>: The List of Datacenters available
    */
+  @ApiOperation(value = "Retrieving all Vim Instances", notes = "")
```
Comment (lorenzotomasini) [inline api/src/main/java/org/openbaton/nfvo/api/admin/RestVimInstances.java]: Why no notes here? is it something like description?

File: api/src/main/java/org/openbaton/nfvo/api/admin/RestVimInstances.java [hunk 4]
```diff
@@ -167,6 +176,7 @@ public VimInstance update(
    * @param idImage of {@code NFVImage}
    * @return {@code NFVImage} selected
    */
+  @ApiOperation(value = "Returns the data about an Image from the Vim Instance", notes = "")
```
Comment (lorenzotomasini) [inline api/src/main/java/org/openbaton/nfvo/api/admin/RestVimInstances.java]: notes is missing sometimes, is there a specific reason?

File: main/src/test/java/org/openbaton/nfvo/core/test/IntegrationClassSuiteTest.java [hunk 5]
```diff
@@ -30,12 +30,14 @@
...
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+import org.springframework.test.context.web.WebAppConfiguration;
 
 /** Created by lto on 20/04/15. */
 @RunWith(SpringJUnit4ClassRunner.class)
 @TestExecutionListeners({DependencyInjectionTestExecutionListener.class})
 @ContextConfiguration(classes = {Application.class})
 @TestPropertySource(properties = {"timezone = GMT", "port: 4242"})
+@WebAppConfiguration
```
Comment (lorenzotomasini) [inline main/src/test/java/org/openbaton/nfvo/core/test/IntegrationClassSuiteTest.java]: Is this needed in the test class by the swagger plugin to work?

File: api/src/main/java/org/openbaton/nfvo/api/admin/RestVimInstances.java (hunk 3 above)
Comment (radoslavvlaskovski) [inline api/src/main/java/org/openbaton/nfvo/api/admin/RestVimInstances.java]: Yes. Notes are a more detailed description and I did not know what more to add here.

Comment (lorenzotomasini) [inline api/src/main/java/org/openbaton/nfvo/api/admin/RestVimInstances.java]: "This method returns the list of all the VimInstances on-boarded in this project identified by the header project-id"

File: api/src/main/java/org/openbaton/nfvo/api/admin/RestKeys.java (hunk 2 above)
Comment (radoslavvlaskovski) [inline api/src/main/java/org/openbaton/nfvo/api/admin/RestKeys.java]: I was not able to find a built in solution.

Comment (lorenzotomasini) [inline api/src/main/java/org/openbaton/nfvo/api/admin/RestKeys.java]: ok for this release. we will find a more dynamic solution in the next

**Comment by radoslavvlaskovski** [comment]
@lorenzotomasini the notes have been updated

**Comment by lorenzotomasini** [comment]
what is missing?

**Pull request merged**